app/src/main/java/com/canvearth/canvearth/pixel/Color.java

```diff
@@ -35,31 +35,46 @@ public Color clone() throws CloneNotSupportedException {
         return color;
     }
 
-    public boolean isTransparent() {
+    public boolean equals(Color color) {
+        return this.r.longValue() == color.r.longValue()
+                && this.g.longValue() == color.g.longValue()
+                && this.b.longValue() == color.b.longValue()
+                && this.a.longValue() == color.a.longValue();
+    }
+
+    public boolean transparent() {
         return (this.a < Constants.COLOR_TRANSPARENT_BOUND) ;
     }
 
     public String toString() {
         return "rgba(" + r + ", " + g + ", " + b + ", " + a + ")";
     }
 
+    // TODO consider this one more time
     public void replaceColorPortion(Color originColor, Color newColor, double portion) {
```

Comment:
What does this method do? Is originColor needed?

Reply:
This method have to be improved.
I'll explain this method's role.
Let's consider the situation when parent pixel's color A is composed of its children's color B,C,D,E.
When B is changed to B', A have to be changed accordingly.
In this situation, we have to call `A.replaceColorPortion(B, B', 0.25)` (0.25 is because B contributes 0.25 for A)

Reply:
Hmm.. Okay, how about adding a short comment about the method usage as it is hard to understand without any context.

Reply:
Got it